lib/sdr_client/redesigned_client/deposit.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module SdrClient
+  class RedesignedClient
+    # Deposit into the SDR API
+    class Deposit
+      def self.deposit_model(...)
+        new(...).deposit_model
+      end
+
+      # @param [Cocina::Model::RequestDRO] model for depositing
+      # @param [Boolean] accession should the accessionWF be started
+      # @param [String] basepath filepath to which filepaths are relative
+      # @param [Array<String>] files a list of relative filepaths to upload
+      # @param [Hash] options optional parameters
+      # @option options [Boolean] assign_doi should a DOI be assigned to this item
+      # @option options [String] priority what processing priority should be used ('low', 'default')
+      # @option options [String] grouping_strategy what strategy will be used to group files
+      # @option options [String] file_set_strategy what strategy will be used to group file sets
+      # @option options [RequestBuilder] request_builder a request builder instance
+      def initialize(model:, accession:, basepath:, files: [], **options)
+        @model = model
+        @accession = accession
+        @basepath = basepath
+        @files = files
+        @options = options
+      end
+
+      def deposit_model # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+        check_files_exist!
+        child_files_match! unless options[:request_builder]
+
+        file_metadata = UploadFilesMetadataBuilder.build(files: files, mime_types: mime_types, basepath: basepath)
+        upload_responses = UploadFiles.upload(file_metadata: file_metadata,
+                                              filepath_map: filepath_map)
+        if options[:request_builder]
+          @model = StructuralGrouper.group(
+            request_builder: options[:request_builder],
+            upload_responses: upload_responses,
+            grouping_strategy: options[:grouping_strategy],
+            file_set_strategy: options[:file_set_strategy]
+          )
+          child_files_match!
+        end
+
+        new_request_dro = UpdateDroWithFileIdentifiers.update(request_dro: model,
+                                                              upload_responses: upload_responses)
+        CreateResource.run(accession: accession,
+                           priority: options[:priority],
+                           assign_doi: options[:assign_doi],
+                           metadata: new_request_dro)
+      end
+
```

Comment:
This method is a bit messy because it's taking the place of two separate classes in the old design. Further internal refactoring would be good after we switch to the new public interface.